src/chain/sync/grandpa_warp_sync.rs

```diff
@@ -84,25 +87,46 @@ pub enum GrandpaWarpSync<TSrc> {
 impl<TSrc> GrandpaWarpSync<TSrc> {
     fn from_babe_fetch_epoch_query(
         query: babe_fetch_epoch::Query,
-        fetched_current_epoch: Option<BabeEpochInformation>,
+        fetched_current_epoch: Option<PartialBabeEpochInformation>,
         state: PostVerificationState<TSrc>,
     ) -> Self {
         match (query, fetched_current_epoch) {
             (babe_fetch_epoch::Query::Finished(Ok((next_epoch, runtime))), Some(current_epoch)) => {
-                let slots_per_epoch = match state.start_chain_information.consensus {
-                    ChainInformationConsensus::Babe {
-                        slots_per_epoch, ..
-                    } => slots_per_epoch,
-                    _ => unreachable!(),
-                };
+                let (slots_per_epoch, babe_config_c, babe_config_allowed_slots) =
+                    match state.start_chain_information.consensus {
+                        ChainInformationConsensus::Babe {
+                            slots_per_epoch,
+                            finalized_next_epoch_transition,
+                            ..
+                        } => (
+                            slots_per_epoch,
+                            finalized_next_epoch_transition.c,
+                            finalized_next_epoch_transition.allowed_slots,
```

Comment:
```suggestion
                            // TODO: /!\ /!\ shouldn't take the same configuration as the genesis; this is a hack while waiting for https://github.com/paritytech/substrate/issues/8060
                            finalized_next_epoch_transition.c,
                            finalized_next_epoch_transition.allowed_slots,
```

Reply:
Please document or something

Reply:
Basically, someone else than you and me should be able to read the code and figure out how it works